script.js

```diff
@@ -201,12 +221,21 @@ const createTodoElement = (todo) => {
     return $todo;
 };
 
-const renderTodos = (todos) => {
+const renderTodos = (stateValue) => {
```

Comment:
Bhai if we do not accept "stateValue" in render function, we have to create another function to update the global filter state value. Or another way to conditionaly update state value in the render function based on the calling of renderTodo function while clicking the filter buttons.

Reply:
> we have to create another function to update the global filter state value.
This is called a mutation or reducer function. 
It's good pattern because it's supposed to be pure and effect free

User clicks on a filter button (let's say complete) -> call setFilter("complete") -> call renderTodos()
Inside renderTodos we'll have the latest value of global `filterState`